website/identifiers/clients/crossref.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+import furl
+import lxml
+import time
+import logging
+
+import requests
+
+from website.identifiers.metadata import remove_control_characters
+from website.identifiers.clients.base import AbstractIdentifierClient
+from website import settings
+
+logger = logging.getLogger(__name__)
+
+CROSSREF_NAMESPACE = 'http://www.crossref.org/schema/4.4.1'
+CROSSREF_SCHEMA_LOCATION = 'http://www.crossref.org/schema/4.4.1 http://www.crossref.org/schemas/crossref4.4.1.xsd'
+CROSSREF_ACCESS_INDICATORS = 'http://www.crossref.org/AccessIndicators.xsd'
+CROSSREF_RELATIONS = 'http://www.crossref.org/relations.xsd'
+CROSSREF_SCHEMA_VERSION = '4.4.1'
+JATS_NAMESPACE = 'http://www.ncbi.nlm.nih.gov/JATS1'
+XSI = 'http://www.w3.org/2001/XMLSchema-instance'
+CROSSREF_DEPOSITOR_NAME = 'Open Science Framework'
+
+
+class CrossRefClient(AbstractIdentifierClient):
+
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+    def build_doi(self, preprint):
+        from osf.models import PreprintProvider
+
+        prefix = preprint.provider.doi_prefix or PreprintProvider.objects.get(_id='osf').doi_prefix
+        return settings.DOI_FORMAT.format(prefix=prefix, guid=preprint._id)
+
+    def build_metadata(self, preprint, status='public', **kwargs):
+        """Return the crossref metadata XML document for a given preprint as a string for DOI minting purposes
+
+        :param preprint -- the preprint, or list of preprints to build metadata for
```

Comment:
Veeeery minor: Sphinx syntax uses `:`, not `--`. So this would be

```
:param PreprintService|list preprint: Preprint or list of preprints to build metadata for.
```